cbcontainers/state/components/enforcer_deployment.go

```diff
@@ -80,7 +80,10 @@ func (obj *EnforcerDeploymentK8sObject) MutateK8sObject(k8sObject client.Object,
 	deployment.Spec.Template.ObjectMeta.Labels = desiredLabels
 	deployment.Spec.Template.Spec.ServiceAccountName = commonState.EnforcerServiceAccountName
 	deployment.Spec.Template.Spec.PriorityClassName = commonState.DataPlanePriorityClassName
-	deployment.Spec.Template.Spec.ImagePullSecrets = []coreV1.LocalObjectReference{{Name: commonState.RegistrySecretName}}
+	deployment.Spec.Template.Spec.ImagePullSecrets = getSharedImagePullSecrets(agentSpec)
```

Comment:
this is great, but the only thing that worries me is that the order of the values in the array can change after deploying the component (it happened to us when dealing with arrays).
This will cause an infinite loop of reconciling.

Instead of pushing directly to  `deployment.Spec.Template.Spec.ImagePullSecrets`
we should create a temp array called `desiredImagePullSecrets`
and push there the the shared secrets + the one for the specific image like you are doing now.

After that, we need another general method, that compares arrays by size and values (but doesn't care about order)
we may already have something similar somewhere the env builder code, cause thats where we had a problem in the past.

P.S
If you change `getSharedImagePullSecrets` to `getImagePullSecrets` with a "..." param of `customImagePullSecrets` (similar to what we do with env vars), you can even code reuse more code (the other loop)

Reply:
done in https://github.com/octarinesec/octarine-operator/pull/128/commits/5cbb994fd42c4c4c6e0266a5f269edc3ac4ffa75